langchain/chains/base.py

```diff
@@ -115,20 +115,52 @@ def __call__(
         else:
             return {**inputs, **outputs}
 
+
+class SingleInputChain(Chain, BaseModel, ABC):
+    """Base interface for chains that take in a single input."""
+
+    input_key: str = Field(default="input")  #: :meta private:
+    output_key: str = Field(default="output")  #: :meta private:
+
+    @property
+    def input_keys(self) -> List[str]:
+        """Input keys this chain expects."""
+        return [self.input_key]
+
+    @property
+    def output_keys(self) -> List[str]:
+        """Output keys this chain expects."""
+        return [self.output_key]
+
+    def apply(self, input_list: List[str]) -> List[str]:
+        """Call the chain on all inputs in the list."""
+        return [self({self.input_key: i})[self.output_key] for i in input_list]
+
+    def run(self, text: str) -> str:
+        """Run text in, text out (if applicable)."""
+        return self({self.input_key: text})[self.output_key]
+
+
+class MultiInputChain(Chain, BaseModel, ABC):
+    """Base interface for chains that take in multiple inputs."""
+
+    input_keys: List[str]
```

Comment:
idk if i would add this - LLMChain will be a multi input chain i believe, but the input keys should be derived from the prompt

Reply:
Yeah you're right, I'll leave `input_keys` and `output_keys` as abstract here.